src/asm/section.cpp

```diff
@@ -303,6 +304,8 @@ static struct Section *createSection(char const *name, enum SectionType type,
 static struct Section *getSection(char const *name, enum SectionType type, uint32_t org,
 				  struct SectionSpec const *attrs, enum SectionModifier mod)
 {
+	assert(alignment < 16 && alignOffset < 1 << alignment); // Should be ensured by the caller
```

Comment:
`getSection` is only called by `sect_NewSection` and `sect_SetLoadSection`, and both of those are only called in parser.y, after parsing a `sectattrs`, which may have parsed an `align_spec`, which does its own bounds checking and error-messaging. (A couple other functions in section.cpp already have this assertion.)